### DISCUSSION TITLE
Huggingface TGI / external API Support?

### DISCUSSION THREAD

**Discussion by ibndias** [outlines-dev/outlines#790] [Q&A]
Hi how do i use external HuggingFace TGI as model?

**Comment by maziyarpanahi** [comment]
good question! TGI has now OpenAI support, can it be used that way?

**Reply by rlouf** [reply]
Yes, it uses Outlines under the hood!

**Comment by maziyarpanahi** [comment]
I am coming back to this @rlouf since now the core has been written in Rust. Is it possible to use outlines outside the TGI by somehow connecting to it or its OpenAI compatible endpoint? The one that comes under the hood is pretty limited and not flexible.
